File: back-end/src/main/java/com/tfg/back/model/dtos/client/ClientDtoGet.java
package com.tfg.back.model.dtos.client;

import com.tfg.back.enums.BloodType;
import com.tfg.back.enums.MembershipLevel;
import com.tfg.back.model.EmergencyContact;
import com.tfg.back.model.Notification;
import com.tfg.back.model.dtos.labResults.LabResultDtoGet;
import com.tfg.back.model.dtos.appointment.AppointmentDtoGet;
import com.tfg.back.model.dtos.feedBack.FeedBackDtoGet;
import com.tfg.back.model.dtos.medicalPrescription.MedicalPrescriptionDtoGet;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ClientDtoGet {
    private Long id;
    private String fullName;
    private String email;
    private String phoneNumber;
    private LocalDate dateOfBirth;
    private MembershipLevel membershipLevel;
    private EmergencyContact emergencyContact;
    private LocalDateTime createdAt;
    private List<AppointmentDtoGet> appointments;
    private String address;
    private BloodType bloodType;
    private List<Notification> notifications;
    private List<MedicalPrescriptionDtoGet> prescriptions;
    private List<LabResultDtoGet> labResults;
    private List<FeedBackDtoGet> feedbacksWritten;
}

File: front-end/src/app/models/client-dto-get.ts

import { Feedback } from "../doctor/doctor-dashboard/doctor-dashboard.component";
import { AppointmentDtoGet } from "./appointment-dto-get";
import { EmergencyContact } from "./emergency-contact";
import { BloodType } from "./enums/blood-type";
import { MembershipLevel } from "./enums/membership-level";
import { LabResultDtoGet } from "./lab-result-dto-get";
import { MedicalPrescriptionDtoGet } from "./medical-prescription-dto-get";
import { NotificationDto } from "./notification-dto";

export class ClientDtoGet {
  id!: number;
  fullName!: string;
  email!: string;
  phoneNumber!: string;
  dateOfBirth!: string;
  membershipLevel!: MembershipLevel;
  emergencyContact!: EmergencyContact;
  createdAt!: string;
  address!: string;
  bloodType!: BloodType;

  constructor(init?: Partial<ClientDtoGet>) {
    Object.assign(this, init);
  }
}